### PR TITLE
Heavily cache location type by user for ICDS UCR

### PR DESCRIPTION
**Pull request by emord**
@snopoke @czue from email. Quickly tested locally and it works as expected, but will need to go back through the other UCRs to change it (will likely test on india as well before rolling out to icds)

buddy @proteusvacuum

### CODE REVIEW

**Comment by emord** [comment]
Also FYI @esoergel we had talked a bit about this at DSI I think

**Review by snopoke — APPROVED** [review]

**Review by czue — APPROVED** [review]

**Comment by emord** [comment]
I de-scoped this to only cache the user's location id. I decided caching location tree lookups should be done separately with ancestor_location so that case UCRs can also benefit for that. ( the original email was just to get couch lookups cached and locations are in sql)

File: custom/icds_reports/ucr/expressions.py [hunk 1]
```diff
@@ -314,6 +317,36 @@ class BooleanChoiceQuestion(JsonObject):
     nullable = BooleanProperty(default=True)
 
 
+icds_ucr_quickcache = get_django_quickcache(memoize_timeout=60, timeout=60 * 60)
+
+
+@icds_ucr_quickcache(('user_id',))
+def _get_user_location_id(user_id):
+    user = CommCareUser.get(user_id)
```
Comment (esoergel) [inline custom/icds_reports/ucr/expressions.py]: nit: `CommCareUser.get_by_user_id` seems to be the preferred way to lookup users, although it probably doesn't make a difference.  That method is cached, though.

**Review by esoergel — APPROVED** [review]

File: custom/icds_reports/ucr/expressions.py (hunk 1 above)
Comment (emord) [reply custom/icds_reports/ucr/expressions.py]: Good point. I'm caching this quite a bit more aggressively (60 min vs 5). and now that I think about it, I shouldn't be using this method anyways as it wraps the doc which isn't strictly required.

Going to see if this helps couch for the next couple of days and then will follow up with getting this without wrapping.

**Review by snopoke — COMMENTED** [review]
:rocket:

**Comment by czue** [comment]
:running_man:

File: custom/icds_reports/ucr/expressions.py (hunk 1 above)
Comment (esoergel) [reply custom/icds_reports/ucr/expressions.py]: Oh that's a good point - could probably save a fair bit of processing time here if you skip the `.wrap()` call.

**Pull request merged**